roofit/roofitcore/inc/RooArgSet.h

```diff
@@ -75,35 +75,75 @@ class RooArgSet : public RooAbsCollection {
+  RooAbsCollection(name)
+  {
+    for (const auto elm : inputs) {
+      add(elm);
+    }
+  }
+
 
   virtual ~RooArgSet();
-  // Create a copy of an existing list. New variables cannot be added
-  // to a copied list. The variables in the copied list are independent
-  // of the original variables.
+  /// Create a copy of an existing list. New variables cannot be added
+  /// to a copied list. The variables in the copied list are independent
+  /// of the original variables.
   RooArgSet(const RooArgSet& other, const char *name="");
   virtual TObject* clone(const char* newname) const { return new RooArgSet(*this,newname); }
   virtual TObject* create(const char* newname) const { return new RooArgSet(newname); }
   RooArgSet& operator=(const RooArgSet& other) { RooAbsCollection::operator=(other) ; return *this ;}
 
-  virtual Bool_t add(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add all elements in list to collection
-    return RooAbsCollection::add(col, silent);
-  }
-  virtual Bool_t addOwned(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add all elements in list as owned components to collection
-    return RooAbsCollection::addOwned(col, silent);
-  }
-  virtual void addClone(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add owned clone of all elements of list to collection
-    RooAbsCollection::addClone(col, silent);
-  }
+  using RooAbsCollection::add;
+  virtual Bool_t add(const RooAbsArg& var, Bool_t silent=kFALSE);
 
-  virtual Bool_t add(const RooAbsArg& var, Bool_t silent=kFALSE) ;
+  using RooAbsCollection::addOwned;
   virtual Bool_t addOwned(RooAbsArg& var, Bool_t silent=kFALSE);
-  virtual RooAbsArg *addClone(const RooAbsArg& var, Bool_t silent=kFALSE) ;
+
+  using RooAbsCollection::addClone;
+  virtual RooAbsArg *addClone(const RooAbsArg& var, Bool_t silent=kFALSE);
 
   using RooAbsCollection::operator[];
-  RooAbsArg& operator[](const char* name) const;
+  /// An "array operator". The named element must exist in the set, otherwise
+  /// code will abort.
+  ///
+  /// When used as an lvalue in assignment operations, the element contained in
+  /// the ArgSet will not be changed. The assignment will operate on the referenced element.
+  ///
+  /// Using RooAbsCollection::operator[], one can also access by index.
+  /// Note that the latter returns a pointer for backward compatibility.
+  RooAbsArg& operator[](const std::string& name) const {
+    return operatorBracketImpl(name.c_str());
+  }
+  ///\copydoc RooArgSet::operator[](const std::string&) const
+  RooAbsArg& operator[](const TString& name) const {
+    return operatorBracketImpl(name.Data());
+  }
+
+  ///\copydoc RooArgSet::operator[](const std::string&) const
+  // Note: This needs to be a template to hide this overload in favour of operator[](std::size_t).
+  // If this wasn't a template, calling with 0 creates an ambiguity, as it could be a nullptr.
+  template<typename T>
```

Comment:
The doc seems to belong to the operator at line 131? Maybe clarify in the doc that `T` must be `char` (actually I'd change the signature to `RooAbsArg& operator[](const Char* name) const {` and use `Char` as template parameter name to signal that).